### ISSUE TITLE
Ownership of dynamic values

### ISSUE THREAD

**Issue by nameofuser1** [microavia/messgen#17]
Иногда полезно иметь возможность сохранить сообщение с динамическими полями. Есть предложение написать класс обертку над сообщением и сделать специализации функций сериализации парсинга:
```
class Storage<T> {
    T msg;
     MemoryAllocator mem;
}

int parse<Storage<T>> (Storage<T> &msg, MessageInfo &info)  {
}
```

Пока конкретный интерфейс не продумывал, но в планах пофантазировать.

Несколько тонких моментов, о которых можно подумать:
1. В таком случае в общем-то передавать аллокатор/память отдельными аргументами возможно и нет смысла, тк он будет храниться внутри Storage.
2. Можно как-нибудь с оптимизировать вызовы parse без Storage, чтобы не конструировать аллокатор и не копировать на стек никакие данные.

@ygorshkov  @DrTon что думаете? Может есть другие пути?

**Comment by ygorshkov** [comment]
возможно я не так понял постановку задачи, но хранение аллокатора рядом с сообщением ничего не даст, так как аллокатор сам не владеет никокой памятью

**Comment by nameofuser1** [comment]
@ygorshkov да, я не очень ясно выразился. Также я хочу сделать так, чтобы он владел памятью. Примерно как тут, только чтобы конструировать каждый раз не нужно было:
https://github.com/microavia/messgen/blob/master/port/cpp/messgen/stl.h
